server/backend/eth_backend.go

```diff
@@ -46,22 +47,20 @@ type ethRequest struct {
 }
 
 type EthRPC struct {
-	url    string
-	client httpClient
-	Lgr    *zap.Logger
+	url string
+	Lgr *zap.Logger
 }
 
 func NewEthClient(url string, lgr *zap.Logger) *EthRPC {
 	rpc := &EthRPC{
-		url:    url,
-		client: http.DefaultClient,
-		Lgr:    lgr,
+		url: url,
+		Lgr: lgr,
 	}
 
 	return rpc
 }
 
-func (rpc *EthRPC) Call(method string, params ...interface{}) (json.RawMessage, error) {
+func (rpc *EthRPC) Call(ctx context.Context, method string, params ...interface{}) (json.RawMessage, error) {
```

Comment:
@rkononov Is this custom client necessary for some reason? I would think that we could just use `goclient.Client` or `web3.Client`, or at least leverage `rpc.Client` for the internals.

Reply:
AFAIR we used that because original client has lack of some calls (total_supply and some others).
But if we already have them in the client we could use it, that would simplify the code a lot

Reply:
They did a good job of separating responsibilities in to a low level `rpc.Client` used by the high level `ethclient/goclient.Client`, so we could just use the `rpc.Client` directly all the time in place of some of this code, or we could use a higher level client when possible, and still make our own custom calls through the same shared underlying`rpc.Client`.